services/management/ia/alerts_service.go

```diff
@@ -116,12 +116,19 @@ func (s *AlertsService) ListAlerts(ctx context.Context, req *iav1beta1.ListAlert
 				return err
 			})
 			if e != nil {
+				// Only FindRuleByID from transaction above can return codes.NotFound
+				if st, ok := status.FromError(e); ok && st.Code() == codes.NotFound {
+					s.l.Warnf("The related alert rule most likely was removed: %s", st.Message())
+					continue
+				}
+
 				return nil, e
 			}
 
 			template, ok := s.templatesService.getTemplates()[r.TemplateName]
 			if !ok {
-				return nil, status.Errorf(codes.NotFound, "Failed to find template with name: %s", r.TemplateName)
+				s.l.Warnf("Failed to find template with name: %s", r.TemplateName)
+				continue
```

Comment:
This case possible when a rule created from the template that was created by the user as a file under `/srv/ia/templates/`. So when the user removes the template file the associated rule will be broken and the rule file will be removed soon. But the alert will be present for several seconds (alertmanager delay), better just to skip it.